### PR TITLE
implementation the storage

### PR DESCRIPTION
**Pull request by BusyhinD**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CAPACITY_OF_STORAGE = 10;
+    private StorageImpl<K, V>[] storages;
+    private K key;
+    private V value;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]: If you want to do this via a class, create an intrinsic class with key & value fields

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,49 @@
...
+    private int countOfElements;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:10]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,49 @@
...
+
+    public StorageImpl() {
+        storages = new StorageImpl[CAPACITY_OF_STORAGE];
+        countOfElements = 0;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:14]:
```suggestion
```
int is 0 by default

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,49 @@
...
+    }
+
     @Override
     public void put(K key, V value) {
+        boolean isSameKey = false;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:19]: Use return; instead of this variable in the loop

**Review by zachar-chorny — CHANGES_REQUESTED** [review]

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**